velox/type/DecimalUtil.h

```diff
@@ -32,5 +32,48 @@ class DecimalUtil {
   /// Helper function to convert a decimal value to string.
   template <typename T>
   static std::string toString(const T& value, const TypePtr& type);
+
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleWithRoundUp(
+      const TInput inputValue,
+      const int fromPrecision,
+      const int fromScale,
+      const int toPrecision,
+      const int toScale,
+      const bool nullOnFailure) {
+    int128_t rescaledValue = inputValue.unscaledValue();
```

Comment:
With this change, we should be able to remove the operator overloads in the `xxxDecimal.h`